### DISCUSSION TITLE
Why useWatch and watch API behave differently in this case?

### DISCUSSION THREAD

**Discussion by nhanpdnguyen** [react-hook-form/react-hook-form#3390] [Q&A] [answered]
Hi everyone, I've noticed this small differences between the `watch` and `useWatch` when seeing a glitch during the implementation of a draggable list (using `react-beautiful-dnd`).

In the top parent component, I used `useFieldArray` along with form context. Then, inside an inner component, I used `useWatch`/`watch` to watch items in that fields array. 

When moving item in that fields array, the changes reflects to watched items. Both `watch` API and `useWatch` cause 2 re-renders. However, it took `useWatch` the second re-render to get the items ordered right, the first re-render still does not reflect the changes.

I wonder why this happened or I've misused something. Thank you very much.

Here's the codesanbox link: https://codesandbox.io/s/react-hook-form-usefieldarray-forked-qhotk?file=/src/index.js

Step to reproduced:
1. Go to the sandbox link
2. Click 'Append'
3. Clear the console logs for easier to see the next changes
4. Click 'Move'
5. Watch the console logs and see that in the first "Watched by useWatch: ...", the items are not reordered, while in the first "Watched by watch API: ...", the items are re-ordered correctly

**Comment by bluebill1049** [comment, marked as answer]
because `useWatch` need to trigger anorther re-render to get the updated value, next patch will improve on this behaviour.

**Reply by nhanpdnguyen** [reply]
Thank you for the answer
